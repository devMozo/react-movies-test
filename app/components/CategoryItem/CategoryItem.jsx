import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './CategoryItem.scss';

const CategoryItem = ({ name, id }) => {
    return (
        <section className="CategoryItem" data-cy="CategoryItem">
            <Link to={`/category/${id}`} data-cy="CategoryItem-link">
                {' '}
                {name}{' '}
            </Link>
        </section>
    );
};

CategoryItem.propTypes = {
    name: PropTypes.string,
    id: PropTypes.number,
};

export default React.memo(CategoryItem);
