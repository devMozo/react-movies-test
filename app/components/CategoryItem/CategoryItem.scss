$itemsPerRow: 6;
$marginBetweenRow: 11px;

.CategoryItem {
    width: calc(100% * (1 / #{$itemsPerRow}) - #{$marginBetweenRow});
    margin-bottom: 20px;
    box-shadow: 0 0 10px #ccc;
    background-color: #333;
    transition: 0.3s;

    &:hover {
        opacity: 0.6;
    }

    a {
        width: 100%;
        color: white;
        text-decoration: none;
        text-align: center;
        padding: 20px 40px;
        box-sizing: border-box;
        display: block;
    }
}
