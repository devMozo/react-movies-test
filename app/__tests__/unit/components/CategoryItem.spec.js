import React from 'react';
import { shallow } from 'enzyme';
import CategoryItem from '../../../components/CategoryItem/CategoryItem';

describe('<CategoryItem />', () => {
    const PROPS = {
        id: 1234,
        name: 'Mocked Name',
    };

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<CategoryItem {...PROPS} />);
        expect(component).toMatchSnapshot();
    });
});
