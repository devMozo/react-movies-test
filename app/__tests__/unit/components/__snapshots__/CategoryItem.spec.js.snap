// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryItem /> should render correctly in "debug" mode 1`] = `
<section
  className="CategoryItem"
  data-cy="CategoryItem"
>
  <Link
    data-cy="CategoryItem-link"
    to="/category/1234"
  >
     
    Mocked Name
     
  </Link>
</section>
`;
